binderhub/repoproviders.py

```diff
@@ -208,11 +224,17 @@ def get_resolved_ref(self):
         self.record_id = r.effective_url.rsplit("/", maxsplit=1)[1]
         return self.record_id
 
+    async def get_resolved_spec(self):
+        return self.spec
```

Comment:
We should not return the DOI here because the DOI is like `master` or a tag. At different moments in time it can point to different things. In Zenodo this flexibility is used to allow versions. What we should return here is the Zenodo record ID (which is like the commit). However we can't use that spec and feed it back into a URL to launch a new binder (I think). (Probably also applies to the other DOI based providers)

Reply:
Here I am confused. Zenodo DOI contains the record ID, this makes it is different than branch or tag name or no?

Reply:
I was thinking of this test https://github.com/jupyterhub/binderhub/blob/a168d069772012c52f9ac7056ec22d779927ae69/binderhub/tests/test_repoproviders.py#L40-L52 where the spec is `10.5281/zenodo.3242074`. This might (right now) point to a record ID of 3242074 but as I understand it that is an "accident". So in the future the same DOI could point to a different record ID. Most of my knowledge about versioning in Zenodo comes from https://help.zenodo.org/#versioning

Because the record ID that a DOI points to can change it made me think that a DOI is like a tag or branch name. The record ID is like the commit SHA.

Reply:
thanks a lot Tim, now I understand. I just made a new commit to fix this for zenodo and figshare. can you check it?